### ISSUE TITLE
Crash in Box Sdk

### ISSUE THREAD

**Issue by nikz-dev** [box/box-android-sdk#170]
```
java.lang.RuntimeException: Error receiving broadcast Intent { act=android.net.conn.CONNECTIVITY_CHANGE flg=0x8000010 (has extras) } in com.box.androidsdk.content.auth.OAuthActivity$1@41e6bc00
    at android.app.LoadedApk$ReceiverDispatcher$Args.run(LoadedApk.java:765)
    at android.os.Handler.handleCallback(Handler.java:615)
    at android.os.Handler.dispatchMessage(Handler.java:92)
    at android.os.Looper.loop(Looper.java:137)
    at android.app.ActivityThread.main(ActivityThread.java:4867)
    at java.lang.reflect.Method.invokeNative(Native Method)
    at java.lang.reflect.Method.invoke(Method.java:511)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1007)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:774)
    at dalvik.system.NativeStart.main(Native Method)
Caused by: java.lang.NullPointerException
    at com.box.androidsdk.content.auth.OAuthActivity$1.onReceive(OAuthActivity.java:93)
    at android.app.LoadedApk$ReceiverDispatcher$Args.run(LoadedApk.java:755)
    ... 9 more
```

**Comment by ashishparmar** [comment]
java.lang.RuntimeException: Error receiving broadcast Intent { act=android.net.conn.CONNECTIVITY_CHANGE flg=0x8000010 (has extras) } in com.box.androidsdk.content.auth.OAuthActivity$1@41e6bc00
    at android.app.LoadedApk$ReceiverDispatcher$Args.run(LoadedApk.java:765)
    at android.os.Handler.handleCallback(Handler.java:615)
    at android.os.Handler.dispatchMessage(Handler.java:92)
    at android.os.Looper.loop(Looper.java:137)
    at android.app.ActivityThread.main(ActivityThread.java:4867)
    at java.lang.reflect.Method.invokeNative(Native Method)
    at java.lang.reflect.Method.invoke(Method.java:511)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1007)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:774)
    at dalvik.system.NativeStart.main(Native Method)
Caused by: java.lang.NullPointerException
    at com.box.androidsdk.content.auth.OAuthActivity$1.onReceive(OAuthActivity.java:93)
    at android.app.LoadedApk$ReceiverDispatcher$Args.run(LoadedApk.java:755)
    ... 9 more

**Comment by doncung** [comment]
Please try updating to the baymax branch. We plan to merge into master with this soon, and it fixes and optimizes many areas.

**Comment by ashishparmar** [comment]
What is baymax  branch i could not get you can you please explain more ?

**Comment by doncung** [comment]
This is a branch we have been working on for several months now. We plan to merge it into master after we do additional testing.

**Comment by ashishparmar** [comment]
Can you please add the link? and please make it as soon as possible .

**Comment by doncung** [comment]
https://github.com/box/box-android-sdk/tree/baymax

**Comment by ashughes** [comment]
I'm getting this same crash on version 3.0.4 when attempting to authenticate on some devices. It looks like it's been a while since this was talked about. Any update on when this fix will be released?

**Comment by ashughes** [comment]
I fixed this issue with #211. The NPE was due to `getUrl()` returning `null`. I first tried just checking for `null`, but this caused another issue where `startOAuth()` could get called twice (once here and once in `onResume()`). It really seemed like the connectivity listener was not necessary, so I just removed it entirely and everything seems to work fine (and no NPE).

**Comment by doncung** [comment]
Thanks Andrew. The new changes should have fixed this as well.

**Comment by toshGitHubIrv** [comment]
Hi,
We are using the 4.0.6 version of the library and we see this issue, as well, for Android 4.1 - 4.3.  Please let us know if you need additional information or if you'd like us to open a new issue, rather than using this one.

Tested:
Android 7.1 - OK
Android 7.0 - OK
Android 6.0 - OK
Android 5.1 - OK
Android 5.0 - OK
Android 4.4 - OK
Android 4.3 - Crash
Android 4.2 - Crash
Android 4.1 - Crash

Thank you

**Comment by doncung** [comment]
Assuming it's the same as the previous issue once this is merged it should fix the crash.
https://github.com/box/box-android-sdk/pull/256

If you continue experiencing issues after this, please open a new issue and include the logs.

**Comment by toshGitHubIrv** [comment]
Thank you for the information.  This fixes the crash.  With this in place, for these devices, it will show the login screen and an "Authenticating - Please Wait" box on top, blocking text entry.  I'll continue to investigate this and open up a new issue.  Thanks again.
